x-pack/legacy/plugins/maps/public/layers/vector_layer.js

```diff
@@ -481,10 +481,9 @@ export class VectorLayer extends AbstractLayer {
     try {
       startLoading(SOURCE_DATA_ID_ORIGIN, requestToken, searchFilters);
       const layerName = await this.getDisplayName();
-      const {
-        data: featureCollection,
-        meta
-      } = await this._source.getGeoJsonWithMeta(layerName, searchFilters, registerCancelCallback.bind(null, requestToken));
+      const { data: featureCollection, meta } = this._source.isPushedData()
```

Comment:
Can you move `isPushedData()` outside of the try/catch and to the top of `_syncSource` method? startLoading/stopLoading/onLoadError should never be used if a layer has canned data.

`_syncSource` should have 2 paths
1) layer has canned data - just return it, nothing else to do
2) layer has to fetch data - existing path of execution

Reply:
Can you move `this._source.isPushedData()` to `this.hasPushedData()`? The concept of using canned data should only exist on the layer level and not the source level. That way any layer can support canned data that is potentially stored in the saved object